ddapm_test_agent/trace.py

```diff
@@ -702,6 +702,354 @@ class TraceChunkV07(TypedDict):
     return _verify_v07_payload(payload)
 
 
+def decode_v1(data: bytes) -> v04TracePayload:
+    """Decode a v1 trace payload.
+    The v1 format is similar to the v07 format but in an optimized format and with a few changes:
+    - Strings are deduplicated and sent in a "Streaming" format where strings are referred to by their index in a string table
+    - Trace IDs are sent as 128 bit integers in a bytes array
+    - 'meta' and 'metrics' are now sent as typed 'attributes', more similar to how OTLP traces are sent
+    """
+    payload = msgpack.unpackb(data, strict_map_key=False)
+    return _convert_v1_payload(payload)
+
+
+def _get_and_add_string(string_table: List[str], value: Union[int, str]) -> str:
+    if isinstance(value, str):
+        string_table.append(value)
+        return value
+    elif isinstance(value, int):
+        if value >= len(string_table) or value < 0:
+            raise ValueError(f"Value {value} is out of range for string table of length {len(string_table)}")
+        return string_table[value]
+
+
+def _convert_v1_payload(data: Any) -> v04TracePayload:
+    if not isinstance(data, dict):
+        raise TypeError("Trace payload must be a map, got type %r." % type(data))
+
+    string_table: List[str] = [""]  # 0 is reserved for empty string
+
+    v04Payload: List[List[Span]] = []
+
+    for k, v in data.items():
+        if k == 1:
+            raise TypeError("Message pack representation of v1 trace payload must stream strings")
+        elif k > 1 and k < 10:  # All keys from 2-9 are strings, for now we can just build the string table
+            # TODO: In the future we can assert on these keys
+            if isinstance(v, str):
+                string_table.append(v)
+        elif k == 11:
+            if not isinstance(v, list):
+                raise TypeError("Trace payload 'chunks' (11) must be a list.")
+            for chunk in v:
+                v04Payload.append(_convert_v1_chunk(chunk, string_table))
+        else:
+            raise TypeError("Unknown key %r in v1 trace payload" % k)
+    return cast(v04TracePayload, v04Payload)
+
+
+def _convert_v1_chunk(chunk: Any, string_table: List[str]) -> List[Span]:
+    if not isinstance(chunk, dict):
+        raise TypeError("Chunk must be a map.")
+
+    priority, origin, decision_maker = "", "", ""
+    trace_id, trace_id_high = 0, 0
+    meta: Dict[str, str] = {}
+    metrics: Dict[str, MetricType] = {}
+    for k, v in chunk.items():
+        if k == 1:
```

Comment:
constants for these magic numbers would help with legibility, but not a deal breaker from me.